app/routes/url.tsx

```diff
@@ -29,7 +29,11 @@ enum Action {
 }
 
 function textToUrl(text: string) {
-  return text.startsWith("https%3A%2F%2F") ? decodeURIComponent(text) : text;
+  return text.startsWith("https%3A%2F%2F")
```

Comment:
I know you didn't introduce this bug, but I just noted that this won't work for `http` encoded URLs.

Could you please fix it the above issue, and also convert that double ternary into if statement with comments?